subsys/bluetooth/mesh/gen_lvl_srv.c

```diff
@@ -304,9 +304,11 @@ static int bt_mesh_lvl_srv_init(struct bt_mesh_model *model)
 	net_buf_simple_init_with_data(&srv->pub_buf, srv->pub_data,
 				      sizeof(srv->pub_data));
 
-	if (IS_ENABLED(CONFIG_BT_MESH_SCENE_SRV)) {
-		bt_mesh_scene_entry_add(model, &srv->scene, &scene_type, false);
+#ifdef CONFIG_BT_MESH_SCENE_SRV
```

Comment:
We will still need `#ifdef CONFIG_BT_MESH_SCENE_SRV` here and in model's struct to get rid of the pointer if scene is not compiled.

Reply:
I start thinking of rolling the whole change back, instantiating `bt_mesh_scene_entry` in the model's structure instead of using pointer and removing `bt_mesh_scene_entry_type` argument from `bt_mesh_scene_entry_add`. Then we won't need `#ifdef CONFIG_BT_MESH_SCENE_SRV` in any place (except in `bt_mesh_scene_entry` definition). If scene server model is not compiled, `bt_mesh_scene_entry` will be empty. `bt_mesh_scene_entry_add` can be wrapped into `IS_ENABLED`. The downside of this is that customer will need to initialize `struct bt_mesh_scene_entry::type` before calling `bt_mesh_scene_entry`, e.g. using `BT_MESH_SCENE_ENTRY`.

Reply:
Although this would return us back to the original problem with wasteful memory consumption...